src/main/java/nurseybook/model/task/GhostTask.java

```diff
@@ -1,65 +1,184 @@
 
 /**
- * Represents a whether a task is a ghost task or not in the nursey book.
- * GhostTasks are temporary tasks that are not meant to be shown to the user.
- * Guarantees: immutable; is valid as declared in {@link #isValidGhostTask(String)}
+ * Represents a Ghost Task in NurseyBook. Ghost tasks are tasks that are not meant to be shown to the user
+ * as part of the main list of tasks.
  */
-public class GhostTask {
-    public static final String MESSAGE_CONSTRAINTS = "GhostTask can take either 'true' or 'false' as its values and"
-            + "is case-insensitive. It cannot be null.";
+public class GhostTask extends Task {
 
     /**
-     * The first character of the GhostTask must not be a whitespace,
-     * otherwise " " (a blank string) becomes a valid input.
+     * Creates a Ghost Task object.
+     * Ghost Tasks are tasks that are not meant to be shown to the user as part of the main list of tasks.
+     *
+     * @param desc                      the description of the task
+     * @param dt                        the date and time of the task
+     * @param names                     the names of people associated with the task
+     * @param status                    the completion status of the task
+     * @param recurrence                the recurrence type of the task
      */
-    public static final String VALIDATION_REGEX = "^(?i)(true|false)$";
+    public GhostTask(Description desc, DateTime dt, Set<Name> names, Status status,
+                     Recurrence recurrence) {
+        super(desc, dt, names, status, recurrence);
+    }
 
     /**
-     * If a task is a GhostTask, isGhostTask will be true. Else, it will be false.
+     * Copies the task and returns it.
+     *
+     * @return A copy of the current task.
      */
-    public final boolean isGhostTask;
+    @Override
+    public GhostTask copyTask() {
+        Description copyDesc = new Description(getDesc().value);
+        DateTime copyDt = new DateTime(getDateTime().getStringDate(), getDateTime().getStringTime());
+        Set<Name> copyRelatedNames = new HashSet<>();
+        for (Name name : getRelatedNames()) {
+            copyRelatedNames.add(new Name(name.fullName));
+        }
+        Status copyStatus = new Status(getStatus().isDone, getStatus().isOverdue);
+        Recurrence copyRecurrence = new Recurrence(getRecurrence().toString());
+
+        return new GhostTask(copyDesc, copyDt, copyRelatedNames, copyStatus, copyRecurrence);
+    }
 
     /**
-     * Constructs a {@code GhostTask}.
+     * Marks task as done.
      *
-     * @param isGhostTask True if GhostTask; false otherwise.
+     * @return same task object that has been marked as done
      */
-    public GhostTask(String isGhostTask) {
-        requireNonNull(isGhostTask);
-        checkArgument(isValidGhostTask(isGhostTask), MESSAGE_CONSTRAINTS);
-        this.isGhostTask = Boolean.parseBoolean(isGhostTask);
+    @Override
+    public GhostTask markAsDone() {
```

Comment:
Hmm does this mean that it is possible to mark a ghost task as done? From my understanding, some ghost tasks may be shown after a viewSchedule command but they cannot be marked done or overdue. Perhaps it would be better for this overridden method to throw an exception?

Reply:
The implementation of `markAsDone`, `markAsOverdue` and `markAsNotOverdue` are all going to be fixed by @xiongjya when she changes her implementation of overdue. I have just put these as a placeholder for now, in a way that is similar to the previous implementation of tasks.